### PR TITLE
Login&Sign up

### PR DESCRIPTION
**Pull request by MaximusBro**
I created two components Login&Register. 
~~Register has two nested components: RegisterCandidate and RegisterCompany, which depend on the state of registerVariant and are displayed accordingly. In the future, it is better to change it to one component, because they are almost identical~~
**Update:**

- I completely redesigned the Register, now there is one single RegisterForm component that changes depending on the "company" or "candidate" form option. I also connected the libraries: Yup, Formik. 

- Made validation for RegisterForm and LoginForm
Created the components: InputText, InputCheckbox, InputSelect, InputError

- Fixed a bug with the menu in the Header. The problem was that the animation in the mobile version was triggered and changed the styles and logic for the menu on the PC. Now it works fine, but because of the crutch, it can be improved in the future.

### CODE REVIEW

File: resources/js/Components/Header/TopbarArea/MenuArea/MainMenu/index.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import HeaderImg from "/public/assets/images/header1-logo.svg";
 
 interface MainMenuProps {
 	showMenu: boolean;
-	setShowMenu(param1: boolean): void;
+	setShowMenu(param: boolean): void;
```
Comment (am0nshi) [inline resources/js/Components/Header/TopbarArea/MenuArea/MainMenu/index.tsx]: param?

File: package.json [hunk 2]
```diff
@@ -27,8 +27,10 @@
     "dependencies": {
         "@inertiajs/react": "^1.0.6",
         "@vitejs/plugin-react": "^4.0.0",
+        "formik": "^2.3.0",
```
Comment (am0nshi) [inline package.json]: https://inertiajs.com/forms

File: resources/css/style.css [hunk 3]
```diff
@@ -2096,17 +2096,18 @@ header.style-1 .main-menu ul > li.menu-item-has-children::after {
...
 }
 header.style-1 .main-menu ul > li:hover > ul.sub-menu {
   visibility: visible;
   opacity: 1;
+	
   transform: translateY(0);
 }
 header.style-1 .main-menu ul > li::before {
-  content: url(../images/icon/menu-active.svg);
+  /* content: url(/public/images/icon/menu-active.svg); */
```
Comment (am0nshi) [inline resources/css/style.css]: public/ would be root folder, and will be the only available for users

File: resources/js/Components/Header/TopbarArea/MenuArea/MainMenu/MenuList/MenuItem/index.tsx [hunk 4]
```diff
@@ -11,45 +11,71 @@ interface listProps {
...
-					submenu.current,
-					{
-						height: showAccordion ? "auto" : 0,
-						opacity: showAccordion ? 1 : 0
-					},
-					{ duration: 0.2 },
-				)
-			}
-			enterAnimation()
+	/* useEffect(() => {
```
Comment (am0nshi) [inline resources/js/Components/Header/TopbarArea/MenuArea/MainMenu/MenuList/MenuItem/index.tsx]: ?

File: resources/js/Components/Login/LoginForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import EmailIcon from "/public/assets/images/icon/email-2.svg"
+//Formik
+import { Formik, FormikValues } from "formik";
+import { loginSchema } from "@/Validations/loginValidation";
+import InputPassword from "@/Components/InputPassword";
+import InputCheckBox from "@/Components/InputCheckBox";
+
+
+
+const LoginForm = () => {
```
Comment (am0nshi) [inline resources/js/Components/Login/LoginForm/index.tsx:13]: https://inertiajs.com/forms

File: resources/js/Components/Login/LoginForm/index.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+		setEmail(value);
+	}
+
+	const onChangePassword = (event: React.ChangeEvent<HTMLInputElement>): void => {
+		const value = event.target.value;
+		setPassword(value);
+	}
+
+	return (
+		<>
```
Comment (am0nshi) [inline resources/js/Components/Login/LoginForm/index.tsx]: indents with 2 spaces, not tabs

File: resources/js/Components/Register/RegisterForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+// Componetns
+import InputText from "@/Components/InputText";
+import InputSelect from "@/Components/InputSelect";
+import InputPassword from "@/Components/InputPassword";
+import InputCheckBox from "@/Components/InputCheckBox";
+
+//Formik
+import { Formik, FormikValues } from "formik";
+//Schemas
+import { candidateSchema, companySchema } from "@/Validations/registerValidation";
```
Comment (am0nshi) [inline resources/js/Components/Register/RegisterForm/index.tsx]: not needed

File: resources/js/Components/Register/RegisterForm/index.tsx [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+		password: "",
+		confirmPassword: "",
+		agreement: false,
+	});
+	const onSubmitFormik = useCallback((values: FormikValues) => {
+
+		console.log(values)
+	}, [])
+	useEffect(() => {
+		if (variant === "company") {
```
Comment (am0nshi) [inline resources/js/Components/Register/RegisterForm/index.tsx]: const fields = {...}
...
setInitialValues({...fields, companyName: '', ... })

File: resources/js/Validations/RegisterValidation.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import * as Yup from "yup"
+
+export const candidateSchema = Yup.object().shape({
```
Comment (am0nshi) [inline resources/js/Validations/RegisterValidation.ts]: not needed

**Review by am0nshi — CHANGES_REQUESTED** [review]

File: resources/js/Components/Login/LoginForm/index.tsx (hunk 6 above)
Comment (MaximusBro) [reply resources/js/Components/Login/LoginForm/index.tsx]: For some reason, my IDE has 2 spaces, not tabs, but GitHub shows two tabs. 
I will try to fix it somehow

**Pull request merged**